src/python/dxpy/sugar/transfers.py

```diff
@@ -0,0 +1,952 @@
+            directory).
+        project (str): The ID of the project that contains the file, if it is not the
+            currently selected project and is not specified in the remote file
+            object/link.
+        block (bool): Wait for the download to complete before returning.
+
+    Notes:
+        Supported filetypes: *.tar.gz, *.tgz, *.tar, *.tar.bz2, *.tbz2, *.gz, *.bz2.
+
+        The arg skip_unpack only impacts tar files, and the arg skip_decompress only
+        impacts non-tar files of type *.gz.
+
+    Returns:
+        str: filename or local named pipe which file was downloaded to. If `block`
+        is False, also returns the Popen for the process running in the background.
+    """
+    remote_file = _as_dxfile(remote_file, project)
+
+    if remote_filename is None:
+        remote_filename = remote_file.describe()["name"]
+
+    if output_dir:
+        makedirs(output_dir, exist_ok=True)
+    else:
+        output_dir = os.getcwd()
+
+    is_tar = (
+        ".tar" in remote_filename
+        or remote_filename.endswith(".tgz")
+        or remote_filename.endswith(".tbz2")
+    )
+    unpack = is_tar and not skip_unpack
+    unzip = (
+        not skip_decompress
+        and not is_tar
+        and (remote_filename.endswith(".gz") or remote_filename.endswith(".bz2"))
+    )
+
+    if not any((unpack, unzip)):
+        return simple_download_file(remote_file, local_filename, output_dir, block)
+
+    dl_func = download_and_unpack_archive if unpack else download_and_decompress_file
+    return dl_func(remote_file, remote_filename, local_filename, output_dir, block)
+
+
+def simple_download_file(
+    remote_file,
+    local_filename=None,
+    output_dir=None,
+    project=None,
+    block=True,
+    **kwargs
+):
+    """
+    Download a file.
+
+    Args:
+        remote_file (dxpy.DXFile or dxlink): The file to download.
+        local_filename (str): The local_filename, or None to use the input filename.
+        output_dir (str): The output directory, or None to use the current directory.
+        project (str): The ID of the project that contains the file, if it is not the
+            currently selected project and is not specified in the remote file
+            object/link.
+        block (bool): Wait for the download to complete before returning. Ignored if
+            create_named_pipe=True.
+        kwargs: Additional arguments passed to `dxpy.sugar.processing.run_cmd()`.
+
+    Returns:
+        The output filename, if `block is True`, otherwise a
+        :class:`dxpy.sugar.processing.Processes` object.
+    """
+    remote_file = _as_dxfile(remote_file, project)
+
+    if local_filename is None:
+        local_filename = SPECIAL_RE.sub("", remote_file.name)
+    if output_dir:
+        local_filename = os.path.join(output_dir, local_filename)
+
+    LOG.info("Downloading file %s to %s", remote_file.get_id(), local_filename)
+
+    cmd = ["dx", "download", remote_file.get_id(), "-o", local_filename]
```

Comment:
For simple download it would be better to run `dxpy.download_dxfile()`

Reply:
Changed to using download_dxfile when block is True.